### PR TITLE
Address TS warnings in terminal on yarn start

### PR DESCRIPTION
**Pull request by jessethach**
- Resolve linting warnings
- Add comments to supporessed lines until a proper refactor can happen.
- Add to gitignore for docker files runinng on mac m1 Apple Silcon machines.

## Background
Provide a brief overview of the work completed here.

## GitHub Issue
#1589 

## Associated PRs
If applicable, replace this text with links to additional pull requests associated with this work.

## Validation Plan

- [ ] Log in as Admin user and test app wide functionality
- [ ] Log in as Provider user and test app wide functionality

## Tech Specs
n/a

## Automated Testing
- [ ] All unit tests are passing.
- [ ] Unit tests have been added to cover this changeset.

## Additional Context
None

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,4 +50,7 @@ fawkes-bundle.zip
 !.elasticbeanstalk/*.global.yml
 
 # Data manipulation script config file
-src/data/prod/config.json
\ No newline at end of file
+src/data/prod/config.json
+
+# M1 docker config
+docker-compose-m1*
```
Comment (bamader) [inline .gitignore:56]: Does this have any effect on non-M1 users?

**Review by bamader — APPROVED** [review]
Good cleanup overall. Documentation around dependencies that couldn't be added is a nice touch. Looks like local tests are failing just because of snapshots, so that's an easy fix: from the `client/` directory of the project in your terminal, run `yarn run test -u` (or, if you run through docker, `docker-compose exec client yarn run test -u`) to auto update the snapshots.

File: .gitignore (hunk 1 above)
Comment (jessethach) [reply .gitignore:56]: This is purely for m1 users to persist a config that works. I am having to ignore my compose files on commits and stash them when switching between branches. So should have no effect on non m1 users.

**Comment by jessethach** [comment]
@bamader Always forget about those snapshots. Maybe we can add a commit hook to run these tests locally? I will get these resolved and make an update!

**Comment by bamader** [comment]
At one point, there was a team decision to actually _not_ make them a commit hook that auto runs, the theory being that if we're doing something that changes the appearance of the app because of the underlying elements, the dev should have to make a conscious choice to run the tests to avoid anything inadvertent. Frankly, though, in 1.5 years of having these tests, I think they may have caught **one** thing, **once**. So I personally would be a fan of either commit-hook or just ditching them altogether. Perhaps something to consider as one of our overhaul points!

**Review by nickbristow — APPROVED** [review]

**Comment by nickbristow** [comment]
## Validations
- [ ] Admin
- [ ] Single org user
- [ ] Multi org user
- [ ] Upload
- [ ] Create flow

**Pull request merged**